### PR TITLE
Boardsource holiday

### PR DESCRIPTION
**Pull request by daysgobye**
## Description

Adding folder holiday to the Boardsource folder to hold this keyboard and future themed boards without cluttering main Boardsource. I added this keyboard to that folder with 2 keymaps on top of the default to show how to use some of the features.  

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — COMMENTED** [review]
Keyboard add looks good, however the `lib/vusb` changes will have to be reverted for us to be able to merge this PR.

**Comment by daysgobye** [comment]
> Keyboard add looks good, however the `lib/vusb` changes will have to be reverted for us to be able to merge this PR.

Well I dont think I did that in the cleanest way but its done LOL sorry about that.

File: keyboards/boardsource/holiday/spooky/config.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0xFEED
```
Comment (Erovia) [inline keyboards/boardsource/holiday/spooky/config.h:23]:
```suggestion
#define VENDOR_ID    0x4273
```
A number of Boardsource boards already use this VID, so why not here?

**Review by Erovia — CHANGES_REQUESTED** [review]

File: keyboards/boardsource/holiday/spooky/config.h (hunk 1 above)
Comment (daysgobye) [reply keyboards/boardsource/holiday/spooky/config.h:23]: My bad good catch.

File: keyboards/boardsource/holiday/spooky/keymaps/rip_mx/keymap.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        RGB_TOG, KC__VOLUP,  KC_F2,
+        RGB_MOD, KC__VOLDOWN,KC_F1
```
Comment (drashna) [inline keyboards/boardsource/holiday/spooky/keymaps/rip_mx/keymap.c:27]: The default volume keycodes should work on Windows, MacOS and Linux. No need to use the custom ones for mac. 
```suggestion
        RGB_TOG, KC_VOLU, KC_F2,
        RGB_MOD, KC_VOLD, KC_F1
```

Comment (daysgobye) [reply keyboards/boardsource/holiday/spooky/keymaps/rip_mx/keymap.c:27]: my bad thank you

File: keyboards/boardsource/holiday/spooky/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# spooky
+
+![spooky](https://i.imgur.com/qAeA66w.jpg)
```
Comment (fauxpark) [inline keyboards/boardsource/holiday/spooky/readme.md:3]:
```suggestion
![spooky](https://i.imgur.com/qAeA66wl.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/boardsource/holiday/spooky/spooky.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/boardsource/holiday/spooky/spooky.h:29]:
```suggestion
#define LAYOUT_ortho_2x3( \
```

Maybe? :D

**Review by drashna — APPROVED** [review]
Love this!

**Comment by daysgobye** [comment]
> Love this!

Im glad! Sorry it took me so long to see this my bad

**Review by fauxpark — APPROVED** [review]

File: keyboards/boardsource/holiday/spooky/config.h [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x4273
+#define PRODUCT_ID   0x0000
```
Comment (noroadsleft) [inline keyboards/boardsource/holiday/spooky/config.h:24]: :grinning: 

```suggestion
#define PRODUCT_ID   0x1031
```

File: keyboards/boardsource/holiday/spooky/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# spooky
+
+![spooky](https://i.imgur.com/qAeA66wl.jpg)
+
+this is a limited edition macro pad for the spookiest time of year
+
+* Keyboard Maintainer: [boardsource](https://github.com/boardsource)
+* Hardware Supported: v1 pcb
+* Hardware Availability:[From Boardsource store around halloween time](https://boardsource.xyz/store/5f783f6da2c1b43e37ca0795)
```
Comment (noroadsleft) [inline keyboards/boardsource/holiday/spooky/readme.md:9]: Minor tweaks.

```suggestion
* Hardware Supported: spooky v1 PCB
* Hardware Availability: [From Boardsource store around halloween time](https://boardsource.xyz/store/5f783f6da2c1b43e37ca0795)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**